packages/server/src/workers/utils.ts

```diff
@@ -116,6 +116,28 @@ export function isDeleteInteraction(subscription: Subscription): boolean {
   return supportedInteractionExtension?.valueCode === 'delete';
 }
 
+export async function isFhirCriteriaMet(subscription: Subscription, currentResource: Resource): Promise<boolean> {
+  const criteria = getExtension(
+    subscription,
+    'https://medplum.com/fhir/StructureDefinition/fhir-path-criteria-expression'
+  );
+  if (!criteria?.valueString) {
+    return true;
+  }
+  const history = await systemRepo.readHistory(currentResource.resourceType, currentResource?.id as string);
+  const previousResource = history.entry?.[1]?.resource as Resource;
+  // If there is no previous resource, then we can't compare and have the criteria automatically pass
+  if (!previousResource) {
+    return true;
+  }
+  const evalInput = { current: toTypedValue(currentResource), previous: toTypedValue(previousResource) };
+  const evalValue = evalFhirPathTyped(criteria.valueString, [toTypedValue(currentResource)], evalInput);
+  if (evalValue?.[0].value === true) {
```

Comment:
FHIRPath uses a lot of "truthy" behavior.  We export a utility method called `toJsBoolean` which would be good here:

https://github.com/medplum/medplum/blob/main/packages/core/src/fhirpath/utils.ts#L39-L48

```ts
return toJsBoolean(evalValue);
```

Or even:

```ts
return toJsBoolean(evalFhirPathTyped(criteria.valueString, [toTypedValue(currentResource)], evalInput));
```

Reply:
I think there's some benefit in being strict here: if a user gives us an expression that doesn't evaluate to a boolean, we can't really infer their intent and should consider treating that as an error that's reported to the subscriber

Reply:
Sad that the `fhirPathCriteria` spec isn't more clear here.  I don't see any explicit definition.

The FHIRPath `iif` function includes the language "The criterion expression is expected to evaluate to a Boolean.", which is perhaps the best comparable.

https://hl7.org/fhirpath/#iifcriterion-expression-true-result-collection-otherwise-result-collection-collection

So yeah, @jamestouri , go ahead with @mattwiller 's suggestion and keep it as you had it.